specs/cli/plugin-add.md

```diff
@@ -0,0 +1,186 @@
+> [!NOTE] 
+> In one's solution, there might be two different [API Manifests][def]. The `apimanifest.json` in the root folder represents a single artifact surface of all APIs and it will always be generated. The second one, specific to each plugin, will be named `{plugin-name}-apimanifest.json` and saved in the chosen output directory when `apimanifest` value is used as the plugin type.
+
+Once the `workspace.json` file is generated and the OpenAPI description file is saved locally, the generation will be executed and the plugin and the sliced OpenAPI description will become available.
+
+For `openai` plugins, the mapping should follow [Hidi logic to generate OpenAI Plugin](https://github.com/microsoft/OpenAPI.NET/blob/vnext/src/Microsoft.OpenApi.Hidi/OpenApiService.cs#L748). Requiring fields default as the following:
+
+| OpenAI field | Default value |
+| -- | -- |
+| name_for_human | Defaults to the OpenAPI document title. |
+| name_for_model | Defaults to `{name_for_human}`. |
+| description_for_human | Defaults to the description from the OpenAPI document.  If the description is not available, it defaults to `Description for {name_for_human}`. |
+| description_for_model | Defaults to `{description_for_human}`. |
+| contact_email | Defaults to the contact email from the OpenAPI document. If the contact email is not available, it defaults to 'publisher-email@example.com'. |
+| logo_url | Dummy URL? |
+| legal_info_url | Dummy URL? |
+|  |  |
+
+For `apimanifest`, the mapping should follow the [OpenApi.ApiManifest lib map](https://github.com/microsoft/OpenApi.ApiManifest/blob/main/docs/OpenApiToApiManifestMapping.md). Requiring fields are as the following:
+
+| API Manifest field | Default value |
+| -- | -- |
+| apiDependencies.Key | Defaults to the plugin name. |
+| publisherName | Defaults to the contact name from the OpenAPI document. If the contact name is not available, it defaults to 'publisher-name'. |
+| publisherName | Defaults to the contact name from the OpenAPI document. If the contact name is not available, it defaults to 'publisher-name'. |
+| publisherEmail | Defaults to the contact email from the OpenAPI document. If the contact email is not available, it defaults to 'publisher-email@example.com'. |
+|  |  |
+
+## Parameters
+
+| Parameters | Required | Example | Description | Telemetry |
+| -- | -- | -- | -- | -- |
+| `--plugin-name \| --pn` | Yes | GitHub | Name of the plugin. Unique within the parent API. Defaults to `Plugin` | No |
+| `--openapi \| -d` | Yes | https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/api.github.com/api.github.com.json | The location of the OpenAPI description in JSON or YAML format to use to generate the plugin. Accepts a URL or a local directory. | No |
+| `--search-key \| --sk` | No | apisguru::github.com:api.github.com | The search key used to locate the OpenAPI description. | Yes, without its value |
```

Comment:
```suggestion
```

right now, none of the client commands have support for search key. Given our time constraints, I'd argue we remove it for now, and bring it back to the whole CLI surface later if we need to.

Reply:
I was following the same pattern as the client-add spec. Should we them remove this same info from the client-add spec?

Reply:
yes for consistency. We postponed any future search improvement until we have a solution for the registry.
I think the fact that it was still present in the client commands is an oversight.

Reply:
@baywet, we kept it to ensure that we can have full repro on generation if it comes from a search.

Reply:
sure, but last time we discussed about that, the key is not something we store in the config. So in the scenario where the key remains the same, but the underlying description moves, the generation/refresh will be broken.
When we got to that discussion as far as I can remember, we said we're due to improve the search experience anyway, and decided to exclude any search related work for the time being.

Reply:
I'm ok with that. The --search-key was removed from the spec for now.